src/controller/CHIPDeviceController.h

```diff
@@ -374,6 +386,11 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
      */
     CHIP_ERROR Init(NodeId localDeviceId, CommissionerInitParams params);
 
+    /**
+     * @brief
+     *  Tears down the entirety of the stack, including destructing key objects in the system.
+     *  This is not a thread-safe API, and should be called with external synchronization.
+     */
```

Comment:
Maybe explicitly point to `DeviceController::Shutdown` documentation here, since that provides more details?